inc/class-googleanalytics.php

```diff
@@ -0,0 +1,422 @@
+				[
+					'type' => 'boolean',
+					'default' => false,
+				]
+			);
+		}
+	}
+
+	public function bookAnalyticsSettingsInit(): void {
+		$section = 'analytics_settings_section';
+
+		add_settings_section(
+			$section,
+			'',
+			[ $this, 'analyticsSettingsSectionCallback' ],
+			$this->book_page
+		);
+
+		$this->addBookSettingsFields();
+		$this->registerBookSettings();
+	}
+
+	private function registerBookSettings(): void {
+		register_setting(
+			$this->book_page,
+			self::$google_id_v3_option,
+			[
+				'type' => 'string',
+				'default' => '',
+			]
+		);
+		register_setting(
+			$this->book_page,
+			self::$google_id_v4_option,
+			[
+				'type' => 'string',
+				'default' => '',
+			]
+		);
+	}
+
+	private function addBookSettingsFields(): void {
+		$section = 'analytics_settings_section';
+
+		add_settings_field(
+			self::$google_id_v3_option,
+			$this->v3_input_label,
+			[ $this, 'analyticsInputCallback' ],
+			$this->book_page,
+			$section,
+			[
+				'legend' => $this->v3_input_legend,
+				'version' => 3,
+				'for_book' => true,
+			]
+		);
+		add_settings_field(
+			self::$google_id_v4_option,
+			$this->v4_input_label,
+			[ $this, 'analyticsInputCallback' ],
+			$this->book_page,
+			$section,
+			[
+				'legend' => $this->v4_input_legend,
+				'version' => 4,
+				'for_book' => true,
+			]
+		);
+
+	}
+
+	 // @phpcs:disable Pressbooks.Security.EscapeOutput.OutputNotEscaped
+	 // @phpcs:disable Pressbooks.Security.ValidatedSanitizedInput.InputNotSanitized
+	 // @phpcs:disable Pressbooks.Security.ValidatedSanitizedInput.MissingUnslash
+
+	public function analyticsSettingsSectionCallback(): void {
+		echo '<p>' . __( 'Google Analytics settings.', 'pressbooks' ) . '</p>';
+	}
+
+	public function analyticsInputCallback( array $args ): void {
+		$option = self::getGoogleIDSiteOption( $args['version'], $args['for_book'] );
+		$html = '<input type="text" id="ga_' . $args['version'] . '" name="ga_' . $args['version'] . '" value="' . $option . '" />';
+		$html .= '<p class="description">' . $args['legend'] . '</p>';
+		echo $html;
+	}
+
+	public function analyticsBooksAllowedCallback( array $args ): void {
+		$ga_mu_site_specific_allowed = get_site_option( self::$is_allowed_option );
+		$html = '<input type="checkbox" id="' . self::$is_allowed_option . '" name="' . self::$is_allowed_option . '" value="1"' . checked( $ga_mu_site_specific_allowed, '1', false ) . '/>';
+		$html .= '<p class="description">' . $args[0] . '</p>';
+		echo $html;
+	}
+
+	public function displayNetworkAnalyticsSettings(): void {
+		?>
+		<div class="wrap">
+			<h2><?php _e( 'Google Analytics', 'pressbooks' ); ?></h2>
+			<?php
+			$nonce = ( ! empty( $_REQUEST['_wpnonce'] ) ) ? $_REQUEST['_wpnonce'] : ''; // @codingStandardsIgnoreLine
+			if ( ! empty( $_POST ) ) {
+				if ( ! wp_verify_nonce( $nonce, 'pb_network_analytics-options' ) ) {
+					wp_die( 'Security check' );
+				} else {
+					$this->saveNetworkIDOption( 'ga_3', 3 );
```

Comment:
Do we need both parameters? Couldn't we guess/build the key based on the version number?